### ISSUE TITLE
[Bug]: overwriting BottomSheet.onAnimate will reset snap index to 0 on rerender

### ISSUE THREAD

**Issue by Jacky56** [gorhom/react-native-scroll-bottom-sheet#2043]
### Version

v5

### Reanimated Version

v3

### Gesture Handler Version

v2

### Platforms

Android, Web

### What happened?

overwriting `BottomSheet.onAnimate` will reset snap index to 0 on rerender

### Reproduction steps

overwrite `onAnimate` method for `BottomSheet` with whatever then create an event which causes a rerender

### Reproduction sample

https://snack.expo.dev/@kagutron/bottom-sheet---issue-reproduction-template

**Comment by hanwenbo** [comment]
Yes, there has been bug, you need to change the source code, when you operate fast, it will appear, you can prompt the user on the interface, please slow down  ：D

**Comment by IslamRustamov** [comment]
The issue lies in `handleOnAnimate` function being a dependency for `animateToPosition` in @gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx

I didn't manage to find the reason how a dependency change retriggers `animateToPosition` (which retriggers something else and so on). Will look into that if I have time. But please do join me in this bugfixing.

UPDATE:
I found the reason why that is happening. **handleOnAnimate** recreates, which recreates **animateToPosition**, which recreates **handleSnapToIndex**, which is calling **useEffect** on this [line](https://github.com/gorhom/react-native-bottom-sheet/blob/master/src/components/bottomSheet/BottomSheet.tsx#L1862) . Which is basically why this bug happens. Will think of a fix for that.

UPDATE 1:
Here is the fix https://github.com/gorhom/react-native-bottom-sheet/pull/2073

**Comment by IslamRustamov** [comment]
bump

**Comment by gorhom** [comment]
thanks @IslamRustamov for your investigation and proposed fix. This issue should be resolved with the next release

**Comment by SoheilKhorram** [comment]
> thanks [@IslamRustamov](https://github.com/IslamRustamov) for your investigation and proposed fix. This issue should be resolved with the next release

@gorhom on what date will this update be done? when will be the next release

**Comment by gorhom** [comment]
it should be published now https://github.com/gorhom/react-native-bottom-sheet/releases/tag/v5.1.1
